### ISSUE TITLE
Allow parentheses for the returns in ternary with no-extra-parens

### ISSUE THREAD

**Issue by theGirrafish** [eslint/eslint#10596]
**What rule do you want to change?**
[no-extra-parens](https://eslint.org/docs/rules/no-extra-parens)

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings (an allow option)

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```jsx
{!this.state.update ? (
  <Button inverted color='blue' size='small'>Edit</Button>
) : (
  <div>
    <Button inverted color='green' size='small'>Save</Button>
    <Button inverted color='orange' size='small'>Back</Button>
  </div>
)}
```

**What does the rule currently do for this code?**
With the `no-extra-parens` on and the `{"returnAssign": false}` option set, it throws 
 
    1:23  warning  Gratuitous parentheses around expression          no-extra-parens

**What will the rule do after it's changed?**
It'd be useful to be able to put parentheses around the return of the ternary operators since sometimes it spans multiple lines and it makes it clearer.

**Comment by theGirrafish** [comment]
Hi @platinumazure,

Sorry I should've mentioned I also have `{"ignoreJSX": "multi-line"}` set and it still throws that error.

**Comment by theGirrafish** [comment]
I see where the confusion is for me, the first group is considered a 


```jsx
{!this.state.update ? (
  <Button inverted color='blue' size='small'>Edit</Button> // <-- This is considered a single-line
) : (
  <div>
    <Button inverted color='green' size='small'>Save</Button>
    <Button inverted color='orange' size='small'>Back</Button>
  </div>
)}
```

So I guess my question is, why is that considered as a single-line statement?
